### PR TITLE
Charlesmchen/sae refinements

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk 

### EN share extension icon

simulator screen shot - iphone x - 2018-02-09 at 17 31 13

### EN home view

simulator screen shot - iphone x - 2018-02-09 at 17 31 19

### EN find by phone number

simulator screen shot - iphone x - 2018-02-09 at 17 31 23

### EN attachment approval

simulator screen shot - iphone x - 2018-02-09 at 17 31 28

### HE share extension icon

simulator screen shot - iphone x - 2018-02-09 at 17 32 34

### HE home view

simulator screen shot - iphone x - 2018-02-09 at 17 32 42

### HE find by phone number

simulator screen shot - iphone x - 2018-02-09 at 17 32 45

### HE attachment approval

simulator screen shot - iphone x - 2018-02-09 at 17 32 51

### CODE REVIEW

File: Signal/translations/en.lproj/Localizable.strings [hunk 1]
```diff
@@ -1635,6 +1632,9 @@
 /* Shown when trying to share content to a Signal user for the share extension. Followed by failure details. */
 "SHARE_EXTENSION_UNABLE_TO_BUILD_ATTACHMENT_ALERT_TITLE" = "Unable to Prepare Attachment";
 
+/* Title for the 'share extension' view. */
+"SHARE_EXTENSION_VIEW_TITLE" = "Share to Signal";
```
Comment (charlesmchen) [inline Signal/translations/en.lproj/Localizable.strings]: We should discuss what this copy should be.

File: SignalMessaging/attachments/SharingThreadPickerViewController.m [hunk 2]
```diff
@@ -149,6 +149,12 @@ - (void)threadWasSelected:(TSThread *)thread
 
     NSString *_Nullable messageText = [self convertAttachmentToMessageTextIfPossible];
 
+    // Hide the navigation bar before presenting the approval view.
+    //
+    // Note that cancelling in the approval views will dismiss the entire
+    // share extension, so there is no "back" button.
+    self.navigationController.navigationBarHidden = YES;
```
Comment (charlesmchen) [inline SignalMessaging/attachments/SharingThreadPickerViewController.m]: Previously we hid the navigation bar throughout the SAE, but we need to be able to "back" from the "find by phone number" view and its "country code" view.

Now, we hide the navigation bar only during the last step, the approval views.

File: SignalMessaging/contacts/SelectRecipientViewController.m [hunk 3]
```diff
@@ -238,10 +238,10 @@ - (void)updateCountryWithName:(NSString *)countryName
                   callingCode:(NSString *)callingCode
                   countryCode:(NSString *)countryCode
 {
-
     _callingCode = callingCode;
 
-    NSString *title = [NSString stringWithFormat:@"%@ (%@)", callingCode, countryCode.uppercaseString];
+    NSString *titleFormat = ([UIView new].isRTL ? @"(%2$@) %1$@" : @"%1$@ (%2$@)");
```
Comment (charlesmchen) [inline SignalMessaging/contacts/SelectRecipientViewController.m]: Slight LRT/RTL format change.

File: SignalShareExtension/ShareViewController.swift [hunk 4]
```diff
@@ -10,7 +10,7 @@ import SignalServiceKit
 import PromiseKit
 
 @objc
-public class ShareViewController: UINavigationController, ShareViewDelegate, SAEFailedViewDelegate {
+public class ShareViewController: UIViewController, ShareViewDelegate, SAEFailedViewDelegate {
```
Comment (charlesmchen) [inline SignalShareExtension/ShareViewController.swift]: This didn't need to be a navigation view controller.

**Comment by charlesmchen** [comment]
Note that the caption UI isn't showing up in RTL languages.  I'm filing a card.

**Comment by michaelkirk** [comment]
LGTM

**Pull request merged**